src/nightwatch/nightwatch.config.js

```diff
@@ -0,0 +1,80 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
+/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "chromedriver" }] */
+
+import chromedriver from 'chromedriver';
+import webpack from 'webpack';
+import WebpackDevServer from 'webpack-dev-server';
+import seleniumServerStandaloneJar from 'selenium-server-standalone-jar';
+
+let port = 8080;
+
+const nightwatchConfig = (webpackConfig, srcFolders, providedPort) => {
+  if (providedPort) {
+    port = providedPort;
+  }
+
+  const webpackServer = new WebpackDevServer(webpack(webpackConfig), { quiet: true, hot: false, inline: false });
+  const stopWebpackDevServer = done => webpackServer.close(() => done());
+  const startWebpackDevServer = done => webpackServer.listen(port, '0.0.0.0', () => done());
```

Comment:
Yeah, I took a look at this. I found I was able to hook in and search for an unused port and successfully start the webserver, however within the nightwatch configuration, we specify the urlLauncher as `http://localhost:${port}`. At this point, port becomes static to the initial value we assign and if the portFinder increments/assigned a different unused port the tests  will still look for the initial assigned port.

I feel at this point this would be a nice extra since the current configuration giving users the option to specify their desired port number & makes it possible to run multiple instances from the command line.